RecoEgamma/EgammaElectronAlgos/src/PixelHitMatcher.cc

```diff
@@ -106,52 +114,79 @@ PixelHitMatcher::compatibleSeeds
    typedef std::unordered_map<std::pair<const GeomDet*,GlobalPoint>, TrajectoryStateOnSurface> PosTsosAssoc;
    const int charge = int(fcharge) ;
 
-   const double xmeas_phi = xmeas.phi();
+   // auto xmeas_phi = xmeas.barePhi();
+   auto xmeas_r = xmeas.perp();
+   
+   const float phicut = std::cos(2.5);
+
+
   FreeTrajectoryState fts = FTSFromVertexToPointFactory::get(*theMagField, xmeas, vprim, energy, charge);
   PerpendicularBoundPlaneBuilder bpb;
   TrajectoryStateOnSurface tsos(fts, *bpb(fts.position(), fts.momentum()));
   
   std::vector<SeedWithInfo> result ;
   
-  mapTsos_fast_.clear();  
+  //mapTsos_fast_.clear();  
   mapTsos2_fast_.clear();  
-  mapTsos_fast_.reserve(seeds->size()) ;
+  // mapTsos_fast_.reserve(seeds->size()) ;
   mapTsos2_fast_.reserve(seeds->size()) ;
 
+  // std::vector<TrajectoryStateOnSurface> vTsos(theTrackerGeometry->dets().size());
+  // TrajectoryStateOnSurface vTsos[theTrackerGeometry->dets().size()];
+
+  auto ndets = theTrackerGeometry->dets().size();
+  // std::vector<bool> away(ndets,false);
+
+  int iTsos[ndets];
+  for ( auto & i : iTsos) i=-1;
+  std::vector<TrajectoryStateOnSurface> vTsos; vTsos.reserve(seeds->size());
+
   for(const auto& seed : *seeds) {
     hit_gp_map_.clear();
     if( seed.nHits() > 9 ) {
       edm::LogWarning("GsfElectronAlgo|UnexpectedSeed") <<"We cannot deal with seeds having more than 9 hits." ;
       continue;
     }
+    
     const TrajectorySeed::range& hits = seed.recHits();
     // cache the global points
+   
     for( auto it = hits.first; it != hits.second; ++it ) {
       hit_gp_map_.emplace_back(it->globalPosition());      
     }
-    //iterate on the hits    
-    for( auto it1 = hits.first; it1 != hits.second; ++it1 ) {
+
+    //iterate on the hits 
+    auto he =  hits.second -1;   
+    for( auto it1 = hits.first; it1 < he; ++it1 ) {
       if( !it1->isValid() ) continue;
-      const unsigned idx1 = std::distance(hits.first,it1);
+      auto  idx1 = std::distance(hits.first,it1);
       const DetId id1 = it1->geographicalId();
-      const GeomDet *geomdet1 = it1->det();      
+      const GeomDet *geomdet1 = it1->det();
+
+      auto ix1 = geomdet1->gdetIndex();
+      assert(ix1<int(ndets));
+      /*
+      if (away[ix1]) continue;
```

Comment:
it generates regression w/r/t 8_0_0	 in my opinion it is safe and makes	sense 
I propose to leave it and address it together with Egamma folks